keps/sig-node/4817-resource-claim-device-status/README.md

```diff
@@ -0,0 +1,618 @@
+result for troubleshooting purpose and with a Network result in a modern
+standard format (closer to Pod.Status.PodIPs for example) used by 3rd party
+controllers.
+
+For each device, if required, the DRA Driver processing the device allocation
+can report the status of it in the `Status.DeviceStatuses` of the ResourceClaim
+by using the Kubernetes API.
+
+```golang
+// ResourceClaimStatus tracks whether the resource has been allocated and what
+// the result of that was.
+type ResourceClaimStatus struct {
+    ...
+
+    // DeviceStatuses contains the status of each device allocated for this
+    // claim, as reported by the driver. This can include driver-specific
+    // information. Entries are owned by their respective drivers.
+    //
+    // +optional
+    // +listType=map
+    // +listMapKey=driver
+    // +listMapKey=device
+    // +listMapKey=pool
+    DeviceStatuses []AllocatedDeviceStatus `json:"deviceStatuses,omitempty" protobuf:"bytes,4,opt,name=deviceStatuses"`
+}
+
+
+// AllocatedDeviceStatus contains the status of an allocated device, if the
+// driver chooses to report it. This may include driver-specific information.
+type AllocatedDeviceStatus struct {
+    // Driver specifies the name of the DRA driver whose kubelet
+    // plugin should be invoked to process the allocation once the claim is
+    // needed on a node.
+    //
+    // Must be a DNS subdomain and should end with a DNS domain owned by the
+    // vendor of the driver.
+    //
+    // +required
+    Driver string `json:"driver" protobuf:"bytes,2,rep,name=driver"`
+
+    // This name together with the driver name and the device name field
+    // identify which device was allocated (`<driver name>/<pool name>/<device name>`).
+    //
+    // Must not be longer than 253 characters and may contain one or more
+    // DNS sub-domains separated by slashes.
+    //
+    // +required
+    Pool string `json:"pool" protobuf:"bytes,3,rep,name=pool"`
+
+    // Device references one device instance via its name in the driver's
+    // resource pool. It must be a DNS label.
+    //
+    // +required
+    Device string `json:"device" protobuf:"bytes,4,rep,name=device"`
+
+    // Conditions contains the latest observation of the device's state.
+    // If the device has been configured according to the class and claim
+    // config references, the `Ready` condition should be True.
+    //
+    // +optional
+    // +listType=atomic
+    Conditions []metav1.Condition `json:"conditions" protobuf:"bytes,5,rep,name=conditions"`
+
+    // DeviceInfo contains arbitrary driver-specific data.
+    //
+    // +optional
+    // +listType=atomic
+    DeviceInfo []runtime.RawExtension `json:"deviceInfo,omitempty" protobuf:"bytes,6,rep,name=deviceInfo"`
+
+    // NetworkDeviceInfo contains network-related information specific to the device.
+    //
+    // +optional
+    NetworkDeviceInfo NetworkDeviceInfo `json:"networkDeviceInfo,omitempty" protobuf:"bytes,7,rep,name=networkDeviceInfo"`
```

Comment:
```suggestion
    NetworkInfo NetworkInfo `json:"networkInfo,omitempty" protobuf:"bytes,7,rep,name=networkInfo"`
```
What people think?

Reply:
I think this makes sense, do you think the `DeviceInfo` field should be renamed to `Info`?

Reply:
I think "Opaque" is the word we use elsewhere for RawExtension from fields. That makes it clear only the driver should have expectations about what is in there.

Reply:
In the device configuration, `Opaque` is a combination of the driver + parameters:
https://github.com/kubernetes/api/blob/v0.31.1/resource/v1alpha3/types.go#L579

Reply:
Oh, right, it's the name of the one-of. Can you check other RawExtension fields in the API? Is there any de facto convention?

Reply:
I have only seen the usage of RawExtension called `Data` in [apps/v1](https://github.com/kubernetes/api/blob/v0.31.1/apps/v1/types.go#L953)

Otherwise, there are some other usages of it called `Object`, `Parameters`, `Items`, `Config`, `Extension`, `Args`, `Data`

Reply:
Ok. `Data` here is better than `Info`, IMO.

Reply:
Nobody seemed against it, so I renamed the field and struct `NetworkDeviceInfo` to `NetworkDeviceData` and the field `Info` to `Data` as suggested.

Reply:
sgtm